vendor/views/mixin.py

```diff
@@ -37,23 +37,23 @@ def user_has_product(self):
 
         TODO: move this to some kind of caching or session variable to avoid hitting the DB on every request.
         """
-
+        site = Site.objects.get_current()
         if self.request.user.is_anonymous:
             self.product_owned = False
         else:
             products = self.get_product_queryset()
-            self.product_owned = self.request.user.customer_profile.filter(site=Site.objects.get_current()).get().has_product(products)
+            self.product_owned = self.request.user.customer_profile.filter(site=site).get().has_product(products)
 
         return self.product_owned
 
     def get_product_queryset(self):
         """
         Method to get the Product(s) needed for the check.  Can be overridden to handle complex queries.
         """
-
+        site = Site.objects.get_current_site()
```

Comment:
Site.objects.get_current() is dependent on the SITE_ID to work.

Reply:
Added SetSiteToRequestMixin